### PR TITLE
In updateMemberData(), includes correct file for alert_count()

### LINKED ISSUE #5176: Alerts are broke
I think this says it all really:

https://www.simplemachines.org/community/index.php?topic=563540.0

1. 
2. 

### Environment (complete as necessary)
- Version/Git revision:
- Database Type: mariadb
- Database Version: 
- PHP Version: 7.1 (7.0 at the time of upgrade)

### Additional information/references

### PR DESCRIPTION
**Pull request by Sesquipedalian**
This should fix #5176

### CODE REVIEW

**Comment by Sesquipedalian** [comment]
Please test this to see if it resolves the issue, @smfbigguy.

**Comment by smfbigguy** [comment]
Not that I see.. It creates an error in Profile-modify.php:



    Type of error: Critical
    Error message: Invalid value inserted or no type specified.
    Function: alert_count
    File: /home/****/public_html/Sources/Profile-Modify.php
    Line: 2315
    URL of page causing the error: https://www.whatsurbeef.net/index.phphttps://www.whatsurbeef.net/cron.php

**Comment by smfbigguy** [comment]
OK, I am going to try to explain this. You gave me an edit to put into Subs.php, which I did.  It references Profile-Modify.php This gives errors and alerts do not work. I edited that file awhile ago. When I edit a file I rename the original and leave it on the server. I had another Profile-Modify file named Profile-Modify1.php I used that file and the alerts are working now, sort of. They are working better than before anyway. 
[Profile-Modify1.zip](https://github.com/SimpleMachines/SMF2.1/files/2663499/Profile-Modify1.zip)

**Comment by smfbigguy** [comment]
So I put your edit in the Suns.php but mine looks like this:

include_once($sourcedir . '/Profile-Modify1.php');

**Comment by smfbigguy** [comment]
query_see_board keeps giving errors on line 2315 in Profile-Modify.php

**Comment by smfbigguy** [comment]
Backtrace screenshot

screenshot_2018-12-10 backtrace

**Comment by smfbigguy** [comment]
The screenshot you see is from when I changed things back to the original Profile-Modify.php So it is still erroring.

**Comment by smfbigguy** [comment]
screenshot_2018-12-10 backtrace
Sorry to keep posting. I am getting alerts but not properly. It will show 3 pages of alerts but there are no alerts. When you switch pages it tells you on load of the new page you have no alerts. When an alert comes in you would not know it if you did not check alerts. This concludes my posting for now, lol.

**Comment by Sesquipedalian** [comment]
Thanks, @smfbigguy. Please try again with this latest commit. It should resolve that error message from line 2315.

**Comment by smfbigguy** [comment]
Still gettin errors from that line. 

Invalid value inserted or no type specified.
Function: alert_count

**Comment by Sesquipedalian** [comment]
> Invalid value inserted or no type specified.

I found a bit more broken logic that I think was causing that. Care to try again with the latest commit?

**Comment by smfbigguy** [comment]
This:
https://github.com/SimpleMachines/SMF2.1/pull/5193/files/72b62a376610d707f3802be30d1f9cd86facd966..667891019a02acd7bdc2b5a6dda9610f02cfbc0c

pretty much stopped the errors in the log and now I do have alerts back.

**Comment by smfbigguy** [comment]
Just waitin on some members to show up now...........lmao

**Comment by smfbigguy** [comment]
One problem I see. How do I get rid of alerts that I was getting when this was broken. I keep seeing I have 22 alerts but there are none. Other than this I think it all works fine so far.

**Comment by smfbigguy** [comment]
Just thought I would toss up some screenshots. In this first one it says I have 22 alerts.
alerts

But I don't. I have two.

alerts1

If we go into the alerts it shows 3 pages. But there is not 3 pages. There is only two alerts.

nopages

If we delete the two alerts I do have. all the pages are gone and I have no alerts.

noalerts

Next time I get an alert, it shows I have 22 again.

**Comment by jdarwood007** [comment]
Might need to add a recount function for alerts to the maintenance page.  Kinda outside of the scope of this issue though.

**Comment by smfbigguy** [comment]
Would I be correct in saying that the only way to get the count right would be to go into the database and do some stuff there. ??

**Comment by jdarwood007** [comment]
Receiving a new alert should cause it to recount correctly as well for that member.

**Comment by smfbigguy** [comment]
Well, things are not re-counting correctly as it stands. I guess clearing the table in the database or something might make it work proper. ?? As of right now alerts work fine as far as I am concerned, it's just the count is off.

**Comment by live627** [comment]
Hey,  don't clear the members table! I still want to at your site.

**Comment by smfbigguy** [comment]
Lol, I meant the alerts table or notifications table or whatever it might be called. There has to be away to do it in phpmyadmin but I have not played around in there for awhile.

EDIT: Not doin it tonight either, to late.

File: Sources/Profile-Modify.php [hunk 1]
```diff
@@ -2296,7 +2296,7 @@ function alert_count($memID, $unread = false)
 	$smcFunc['db_free_result']($request);
 
 	// If this isn't the current user, get their boards.
-	if (isset($user_info) && $user_info['id'] != $memID)
+	if (!isset($user_info) || $user_info['id'] != $memID)
```
Comment (albertlast) [inline Sources/Profile-Modify.php:2300]: you access $user_info without knowing if it exists,
this case could happen when this is called by the cronjob i think.

Comment (Sesquipedalian) [reply Sources/Profile-Modify.php:2300]: Actually, that's the opposite of what this does. ;)

Comment (albertlast) [reply Sources/Profile-Modify.php:2300]: hmm because of the "or" php stop the evaluation of the right side because the left side is already true...

Comment (Sesquipedalian) [reply Sources/Profile-Modify.php:2300]: Exactly. :)

**Comment by Sesquipedalian** [comment]
> One problem I see. How do I get rid of alerts that I was getting when this was broken. I keep seeing I have 22 alerts but there are none. Other than this I think it all works fine so far.

I suspect that this is due to a discrepancy between `fetch_alerts()` and `count_alerts()`. Both start out by retrieving all a user's alerts from the database, but then `fetch_alerts()` weeds out any that refer to boards that the user can't see, topics that don't exist, and/or messages that don't exist, while `count_alerts()` only weeds out ones that refer to boards that the user can't see. So if you have alerts in your database that refer to non-existent topics or messages, `count_alerts()` will include them while `fetch_alerts()` won't.

I'll put together another PR to deal with that once this one is merged.

**Comment by smfbigguy** [comment]
Sounds good.

**Comment by smfbigguy** [comment]
I update the files on my site. I think alerts are working now perfectly. My alert count went down.....to 1. I really hated to post this. Please don't open this again if you don't have to. All I am sayin is everything seems to be working. I just have one alert I can not find anywhere.

**Pull request merged**